apps/builder/app/canvas/features/webstudio-component/link.ts

```diff
@@ -0,0 +1,53 @@
+import type { MouseEvent } from "react";
+import { useStore } from "@nanostores/react";
+import { findPageByIdOrPath, Page } from "@webstudio-is/project-build";
+import { pagesStore, useIsPreviewMode } from "~/shared/nano-states";
+import { publish } from "~/shared/pubsub";
+
+declare module "~/shared/pubsub" {
+  export interface PubsubMap {
+    switchPage: { pageId: Page["id"] | "home" };
+  }
+}
+
+const isAbsoluteUrl = (href: string) => {
+  try {
+    new URL(href);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const useHandleLinkClick = () => {
+  const pages = useStore(pagesStore);
+  const [isPreviewMode] = useIsPreviewMode();
+
+  return (event: MouseEvent) => {
+    event.preventDefault();
+
+    if (isPreviewMode === false || pages === undefined) {
+      return;
+    }
+
+    const href = event.currentTarget.getAttribute("href");
+
+    if (href === null) {
+      return;
+    }
+
+    const page = findPageByIdOrPath(pages, href);
+
+    if (page) {
+      publish({
+        type: "switchPage",
+        payload: { pageId: page.id === pages.homePage.id ? "home" : page.id },
+      });
+      return;
+    }
+
+    if (isAbsoluteUrl(href)) {
+      window.open(href, "_blank");
```

Comment:
Does it work from inside iframe? Could be more restrictions in other browsers. Could you check so we could have future proof implementation. Can maybe will be more sense to put href in switchPage event.

Reply:
Tested in Chrome, Brave, Safari, Firefox: all good. Couldn't find anything about it not working via Google.

I think doing it in switchPage will be more risky because then browser may think it's not caused by click.